sklearn/manifold/t_sne.py

```diff
@@ -741,11 +740,11 @@ def _fit(self, X, skip_num_points=0):
                 # the method was derived using the euclidean method as in the
                 # input space. Not sure of the implication of using a different
                 # metric.
-                distances_nn **= 2
+                distances_nn.data **= 2
```

Comment:
Is there a case in which this could be modifying the input `X`? Please double check...

Reply:
`distances_nn` is `X` only with `metric='precomputed'`, not with `metric='euclidean'`.

Reply:
This line is executed only when `self.metric == "euclidean"`